src/core_plugins/tag_cloud/public/tag_cloud.js

```diff
@@ -0,0 +1,250 @@
+import d3 from 'd3';
+import d3TagCloud from 'd3-cloud';
+import vislibComponentsSeedColorsProvider from 'ui/vislib/components/color/seed_colors';
+
+
+const ORIENTATIONS = {
+  'single': (tag) => 0,
+  'rightAngled': (tag) =>~~(Math.random() * 2) * 90,
+  'multi': (tag) => (~~(Math.random() * 12) * 15) - 90
+};
+const D3_SCALING_FUNCTIONS = {
+  'linear': d3.scale.linear(),
+  'log': d3.scale.log(),
+  'sqrt': d3.scale.sqrt()
+};
+
+
+export default class TagCloud {
+
+  constructor(element, size) {
+    this._element = element;
+    this._d3SvgContainer = d3.select(element);
+    this._svgGroup = this._d3SvgContainer.append('g');
+    this._size = [1, 1];
+    this.setSize(size);
+
+    this._fontFamily = 'Impact';
+    this._fontStyle = 'normal';
+    this._fontWeight = 'normal';
+    this._orientations = 'single';
+    this._minFontSize = 10;
+    this._maxFontSize = 36;
+    this._textScale = 'linear';
+    this._padding = 5;
+  }
+
+  setOptions(options) {
+    if (JSON.stringify(options) === this._optionBackup) {
```

Comment:
I think this could use a comment explaining the intention. What's `_optionBackup`?